### PR TITLE
jewel: rgw: add the remove-x-delete feature to cancel swift object expiration

### PR DESCRIPTION
**Pull request by asheplyakov**
http://tracker.ceph.com/issues/19473

### CODE REVIEW

**Comment by smithfarm** [comment]
@rzarzynski This PR is implicated in a "saw valgrind issues" regression http://pulpito.ceph.com/smithfarm-2017-04-27_16:58:41-rgw-wip-jewel-backports---basic-smithi/

The other PRs are the rgw-related ones in this list: http://tracker.ceph.com/issues/19538#note-48

Could you help me analyze and identify the culprit?

**Comment by asheplyakov** [comment]
Jenkins retest this please

**Comment by smithfarm** [comment]
merge conflict

File: src/rgw/rgw_file.cc [hunk 1]
```diff
@@ -1187,8 +1187,8 @@ namespace rgw {
       emplace_attr(RGW_ATTR_SLO_UINDICATOR, std::move(slo_userindicator_bl));
     }
 
-    op_ret = processor->complete(etag, &mtime, real_time(), attrs, delete_at,
-				 if_match, if_nomatch);
+    op_ret = processor->complete(etag, &mtime, real_time(), attrs,
```
Comment (amitkumar50) [inline src/rgw/rgw_file.cc]: @asheplyakov Sorry to Say Alexey, can you please keep title of Future PRs within 50 characters..
https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#title-of-pull-requests-and-title-of-commits

**Comment by smithfarm** [comment]
@rzarzynski This passed an rgw suite at tracker.ceph.com/issues/20613#note-8

Please review.

**Comment by smithfarm** [comment]
@yehudasa @mattbenjamin This PR is backporting a new rgw feature to jewel. Are you OK with that?

**Pull request merged**